tsdb/engine/tsm1/engine.go

```diff
@@ -2507,6 +2546,28 @@ func (e *Engine) createCallIterator(ctx context.Context, measurement string, cal
 		return nil, nil
 	}
 
+	// check for optimized series iteration for tsi index
+	if e.index.Type() == tsdb.TSI1IndexName {
+		indexSet := tsdb.IndexSet{Indexes: []tsdb.Index{e.index}, SeriesFile: e.sfile}
+		seriesOpt := opt
+		if len(opt.Dimensions) == 0 {
+			// no point ordering the series if we are just aggregating them
+			seriesOpt.Ordered = false
```

Comment:
Not sure on this particular check - maybe needs to be more granular depending on the type of call iterator?

Reply:
Yeah, I think there might be some call iterators that need sorting.